tedana/selection/decision_tree_class.py

```diff
@@ -0,0 +1,276 @@
+"""
+Functions that include workflows to identify
+TE-dependent and TE-independent components.
+"""
+
+import inspect
+import json
+import logging
+from pkg_resources import resource_filename
+
+from tedana.selection._utils import (
+    clean_dataframe, confirm_metrics_exist)
+#    are_only_necessary_metrics_used)
+from tedana.selection import selection_nodes
+
+LGR = logging.getLogger(__name__)
+RepLGR = logging.getLogger('REPORT')
+RefLGR = logging.getLogger('REFERENCES')
+
+VALID_TREES = [
+    'mdt', 'minimal_decision_tree1',
+    'kdt', 'kundu_MEICA27_decision_tree'
+]
+
+
+class TreeError(Exception):
+    pass
+
+
+def load_config(tree):
+    """
+    Loads the json file with the decision tree and validates that the
+    fields in the decision tree are appropriate.
+
+    Parameters
+    ----------
+    tree : :obj:`str`
+        A json file name in ./selection/data without the '.json' extension
+
+    Returns
+    -------
+    tree : dict
+        A validated decision tree dictionary
+
+    Note
+    ----
+    In the current version of this script, the decision tree script is validated
+    against a pre-defined list of valid trees. Eventually, there should be a way
+    to load trees that aren't on the validated list and the currently validated
+    list should be used as a short-hand for common trees.
+    """
+
+    fname = resource_filename('tedana', 'selection/data/{}.json'.format(tree))
```

Comment:
I'd prefer to optionally support full paths here, so people can feed in their own trees with a full file path or they can call our built-in trees with just the name of the tree.

Reply:
I just added support for full paths (and changed the error messages accordingly)